### PR TITLE
Document how to enable debug/tracing/... logging in the client - Cheatsheet

### LINKED ISSUE #2526: Output the http REST communication to the log
In order to see messages sent and received from/to the K8s client and debug possible issues, having all the http requests to K8s API logged it's interesting. Having the endpoint, and payload sent/received and http headers.

### PR DESCRIPTION
**Pull request by MUzairS15**
## Description
Updated CHEATSHEET.md
Added example for how to configure okhttp logging-interceptor.
Fixes #2526

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [ ] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: doc/CHEATSHEET.md [hunk 1]
```diff
@@ -2767,3 +2767,38 @@ try (KnativeClient kn = new DefaultKnativeClient()) {
...
+
+- Configure OkHTTP logging:
+``` 
+Logger reqLogger = LoggerFactory.getLogger(HttpLoggingInterceptor.class); 
+```
+ if (reqLogger.isTraceEnabled()) { 
+     HttpLoggingInterceptor loggingInterceptor = new HttpLoggingInterceptor(); 
+     loggingInterceptor.setLevel(HttpLoggingInterceptor.Level.BODY); 
+     httpClientBuilder.addNetworkInterceptor(loggingInterceptor); 
+ }
```
Comment (rohanKanojia) [inline doc/CHEATSHEET.md:2782]: Is this code from [HttpClientUtils ](https://github.com/fabric8io/kubernetes-client/blob/master/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/HttpClientUtils.java#L146)in kubernetes-client? Why do we need this here?

**Comment by MUzairS15** [comment]
can anyone take a look.

File: doc/CHEATSHEET.md [hunk 2]
```diff
@@ -2767,3 +2767,30 @@ try (KnativeClient kn = new DefaultKnativeClient()) {
...
+            .withNewResourceAttributes()
+            .withGroup("apps")
+            .withResource("deployments")
+            .withVerb("create")
+            .withNamespace("rokumar")
+            .endResourceAttributes()
+            .endSpec()
+            .build();
+
+    ssar = client.authorization().v1().selfSubjectAccessReview().create(ssar);
```
Comment (rohanKanojia) [inline doc/CHEATSHEET.md:2794]: I think we can remove this SelfSubjectAccessReview example. You can just show the verbose logs output or maybe just keep it like this:

Using logging-interceptor:

- Set logging level to trace in my simplelogger.properties file would print okhttp requests and response bodies from apiServer:
```
 org.slf4j.simpleLogger.defaultLogLevel=trace
```

Comment (MUzairS15) [reply doc/CHEATSHEET.md:2794]: Untitled
Will this work?

Comment (MUzairS15) [reply doc/CHEATSHEET.md:2794]: Also how i can be added to mailing list?
Or a channel where i can ask questions?

**Comment by MUzairS15** [comment]
Untitled
Will this work?

**Comment by MUzairS15** [comment]
Also how i can be added to a mailing list or a channel where i can resolve my queries?

File: doc/CHEATSHEET.md (hunk 2 above)
Comment (rohanKanojia) [reply doc/CHEATSHEET.md:2794]: I think this should be good enough. However, you have code blocks around line 2776, they should instead be around 2778

**Comment by rohanKanojia** [comment]
You can reach us out here:

Mailing list: fabric8-devclients@googlegroups.com
Gitter channel: https://gitter.im/fabric8io/kubernetes-client

**Comment by rohanKanojia** [comment]
Thanks, you would also need to add this entry in table of contents section at the top

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Review by rohanKanojia — APPROVED** [review]

**Pull request merged**